src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/OsxTcpStatistics.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Net.NetworkInformation
+{
+    internal class OsxTcpStatistics : TcpStatistics
+    {
+        private readonly long _connectionsAccepted;
+        private readonly long _connectionsInitiated;
+        private readonly long _cumulativeConnections;
+        private readonly long _errorsReceived;
+        private readonly long _failedConnectionAttempts;
+        private readonly long _segmentsSent;
+        private readonly long _segmentsResent;
+        private readonly long _segmentsReceived;
+        private readonly int _currentConnections;
+
+        public OsxTcpStatistics()
+        {
+            Interop.Sys.TcpGlobalStatistics statistics;
+            if (Interop.Sys.GetTcpGlobalStatistics(out statistics) != 0)
+            {
+                throw new NetworkInformationException((int)Interop.Sys.GetLastError());
```

Comment:
I think that in the Win32 case the NIE would automatically derive some text for the error codes. It would be nice to have the same behavior for other platforms.